management/src/test/java/org/ehcache/management/providers/statistics/HitCountTest.java

```diff
@@ -100,7 +101,14 @@ public void test() throws InterruptedException, IOException {
       registryConfiguration.addConfiguration(new EhcacheStatisticsProviderConfiguration(1,TimeUnit.MINUTES,100,1,TimeUnit.MILLISECONDS,10,TimeUnit.MINUTES));
       ManagementRegistryService managementRegistry = new DefaultManagementRegistryService(registryConfiguration);
 
-      CacheConfiguration<Long, String> cacheConfiguration = CacheConfigurationBuilder.newCacheConfigurationBuilder(Long.class, String.class, resources).build();
+      CacheConfiguration<Long, String> cacheConfiguration = CacheConfigurationBuilder.newCacheConfigurationBuilder(Long.class, String.class, resources)
+        .withEvictionAdvisor(new EvictionAdvisor<Long, String>() {
+          @Override
+          public boolean adviseAgainstEviction(Long key, String value) {
+            return key.equals(2L);
```

Comment:
> 3 tier test: heap(1, ENTRIES).offheap(2, MB).disk(3, MB)
> 
> cache.put(1L, "1");
> cache.put(2L, "2");
> cache.put(3L, "3");
> -now keys 1,2 and 3 are all in the disk tier.

Correct

> 
> get(1L) 
> -hit on disk tier
> -key 1 is pushed to heap and offheap, and also remains on disk

installed on heap, off heap only misses but does not get the value

> -if evictionAdvisor designated 1L then this key would remain in Heap tier

eviction does not play here, only a single entry exists in heap

> 
> get(2L)
> -hit on disk tier
> -key 2 is pushed to heap and offheap, and also remains on disk

again, only heap is affected, off heap only misses

> -key 1 is removed from heap tier bc only allows 1 entry

since key `2L` is advised against eviction, then key `1L` is the guaranteed eviction victim. Without the eviction advisor, it would be undetermined if both had the same `lastAccess` timestamp, which can happen as failures have shown. Also when they have the same, the eviction victim will really be what is sampled first (or last, would need to check code) and that's _random_.

> -since evictionAdvisor designates 2L it will remain in Heap tier

correct

> 
> get(2L)
> -hit on heap tier
> 
> ***I am assuming at this point key 2L sometimes expires and this was causing the problem? Then by adding the eviction advisor the key no longer can expire and be evicted?

No expiry is at play in this test.

> 
> get(1L)
> -hit on offheap tier
> -Does the evictionAdvisor prevent 1L from being pushed to the heap(heap only allows 1 entry)?

Indeed, it will once again loose the eviction sampling because `2L` is advised against eviction.